lib/action_policy/behaviours/policy_for.rb

```diff
@@ -22,6 +22,10 @@ def authorization_context
         Kernel.raise NotImplementedError, "Please, define `authorization_context` method!"
       end
 
+      private def memoized_authorization_context
+        @_authorization_context ||= authorization_context
+      end
```

Comment:
I moved memoization from `authorization_context` to this private method otherwise it could lead to nonintuitive issues.

My first solution was to merge the explicit context to a newly built one :
```ruby
context = context ? build_authorization_context.merge(context) : authorization_context
```

But then, an overrided `authorization_context` would be ignored:

```ruby
def authorization_context
  { application: current_application_name, user: current_user }
end

def allowed_to_do_something?
  user = find_user

  allowed_to? :do_something?, context: { user: user } 
  # Why application from my authorization_context is not merged ?
end
```